File: backend/src/main/java/com/estacionamiento/backend/service/VehiculoService.java
package com.estacionamiento.backend.service;

import java.util.List;

import com.estacionamiento.backend.dto.BitacoraDTO;
import com.estacionamiento.backend.dto.EntradaVehiculoDTO;
import com.estacionamiento.backend.dto.VehiculoCreateDTO;
import com.estacionamiento.backend.dto.VehiculoGetDTO;
import com.estacionamiento.backend.dto.VehiculoSalidaDTO;

public interface VehiculoService {
    VehiculoGetDTO crearVehiculoOficial(VehiculoCreateDTO vehiculoCreateDTO);
    VehiculoGetDTO crearVehiculoResidente(VehiculoCreateDTO vehiculoCreateDTO);
    VehiculoGetDTO registrarEntradaVehiculo(EntradaVehiculoDTO entradaVehiculoDTO);
    VehiculoSalidaDTO registrarSalidaVehiculo(EntradaVehiculoDTO entradaVehiculoDTO);
    VehiculoGetDTO crearVehiculoNoResidente(VehiculoCreateDTO vehiculoCreateDTO);  
    List<BitacoraDTO> generarInformePagos(); 
    void reiniciarMes();
    List<VehiculoGetDTO> listarVehiculos();
    void procesarPago(String placa);
}

File: frontend/src/app/services/vehiculo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { VehiculoCreateDTO, EntradaVehiculoDTO, VehiculoSalidaDTO, BitacoraDTO } from '../models/vehiculo.model';

interface AppConfig {
  apiUrl: string;
}

@Injectable({
  providedIn: 'root'
})
export class VehiculoService {
  private baseUrl = environment.apiUrl;

  constructor(private http: HttpClient) { }

  registrarOficial(data: VehiculoCreateDTO): Observable<any> {
    return this.http.post(`${this.baseUrl}/vehiculos/oficiales`, data);
  }
  registrarResidente(data: VehiculoCreateDTO): Observable<any> {
    return this.http.post(`${this.baseUrl}/vehiculos/residentes`, data);
  }
  registrarNoResidente(data: VehiculoCreateDTO): Observable<any> {
    return this.http.post(`${this.baseUrl}/vehiculos/no-residentes`, data);
  }
  registrarEntrada(data: EntradaVehiculoDTO): Observable<any> {
    return this.http.post(`${this.baseUrl}/estancias/entrada`, data);
  }
  registrarSalida(data: EntradaVehiculoDTO): Observable<VehiculoSalidaDTO> {
    return this.http.post<VehiculoSalidaDTO>(`${this.baseUrl}/estancias/salida`, data);
  }
  obtenerReporteResidentes(): Observable<BitacoraDTO[]> {
    return this.http.get<BitacoraDTO[]>(`${this.baseUrl}/residentes/pagos`);
  }
  reiniciarMes(): Observable<string> {
    return this.http.post(`${this.baseUrl}/mes/iniciar`, {}, { responseType: 'text' });
  }
  listarVehiculos(): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}/vehiculos`);
  }
  procesarPago(placa: string): Observable<any> {
    return this.http.post(`${this.baseUrl}/vehiculos/${placa}/pagar`, {}, { responseType: 'text' });
  }
}